nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -205,11 +210,21 @@
             .defaultValue(NO_SERVER_SIDE_ENCRYPTION)
             .build();
 
+    public static final PropertyDescriptor OBJECT_TAGS = new PropertyDescriptor.Builder()
```

Comment:
Another option for this property: allow the user to give a regular expression and all the flow file attributes matching the regular expression would be used for tagging. Wouldn't it be more flexible?

Reply:
I don't fully understand this one. Are you suggesting to replace the JSON approach with a different one?

Reply:
Yep, wouldn't it be more flexible?
With the current approach, it assumes that you always have the same set of tags (even if the values can change based on ffs attributes, the keys would always be the same). With the regular expression, you could easily manage the case where you have one ff with attributes tagS3_country=FR, tagS3_security=topsecret and one flow file with only tagS3_country=US. You would set the tag regular expression to tagS3.* and you would have the tags created using the attributes matching the regex. Does it make sense or am I missing something?

Reply:
The keys not necessarily be the same, right? It can change since the property supports expression language so the key-values can be dynamically generated and substituted using expression language. 

Having said that, I do understand your point. My only thought is that when the developer is assigning tags to the FlowFiles, he/she would be using either `UpdateAttribute` or more preferably `ExecuteScript` to do some basic processing and assign tags according to some conditions. Correct? In that case, I honestly don't see the regular expression approach adding more value than the JSON approach. They seem more or less the same. I mean, if the developer is using `ExecuteScript` to generate the tags and add it to FlowFile attributes, he/she might as well generate the entire JSON and assign it to a FlowFile attribute and that FlowFile attribute can be given as the property value here and the substitution can be done. Makes sense?

Reply:
I completely agree that the same result can be achieved with both options. It's just that ExecuteScript will require someone to write some lines of code. Also, I'm not a big fan of the idea to have a big JSON document as a flow file attribute but I don't expect someone to define hundreds of tags. Anyway, I'm ok to merge as-is, I can do it once the other comment is addressed. Thanks!